test/e2e/windows/hybrid_network.go

```diff
@@ -79,6 +72,20 @@ var _ = SIGDescribe("Hybrid cluster network", func() {
 
 		})
 
+		ginkgo.It("should provide Internet connection for containers using DNS [Feature:Networking-DNS][LinuxOnly]", func() {
+			e2eskipper.SkipIfNodeOSDistroIs("windows")
+			ginkgo.By("checking connectivity to 8.8.8.8 53 (google.com) from Linux")
+			framework.ExpectNoError(
+				network.CheckConnectivityToHost(f, "", "connectivity-test", "google.com", 53, 30))
```

Comment:
```suggestion
			assertConsistentConnectivity(f, windowsPod.ObjectMeta.Name, windowsOS, windowsCheck("www.google.com"))
```

Reply:
@aojea I thought that we need to implement this case for linuxOS and the next one for windowOS, you have suggested windowsOS in both cases, why is that the case ?

Reply:
sunday reviews sorry, didn't realize about the OS thing.
is this not a duplicate  of?
> 	ginkgo.It("should provide Internet connection for containers [Feature:Networking-IPv4]", func() {

Reply:
@aojea you suggested that we can add separate test similar to this in your comment https://github.com/kubernetes/kubernetes/issues/107078#issuecomment-997287000

Reply:
ok, I apologize because I should've checked the code better before reviewing.
So, this has a before block
```
	ginkgo.BeforeEach(func() {
		e2eskipper.SkipUnlessNodeOSDistroIs("windows")
	})
```

but then it has a 
```
ginkgo.By("checking connectivity to 8.8.8.8 53 (google.com) from Linux")
			assertConsistentConnectivity(f, linuxPod.ObjectMeta.Name, linuxOS, linuxCheck("8.8.8.8", 53))
```

I recognize I'm not familiar with this hybrid setup, but is the first skip working on hybrid tests?

Reply:
Also can add
**A** : `assertConsistentConnectivity(f, linuxPod.ObjectMeta.Name, linuxOS, linuxCheck("www.google.com"))` instead of **B** : `assertConsistentConnectivity(f, windowsPod.ObjectMeta.Name, windowsOS, windowsCheck("www.google.com"))` because I have added `e2eskipper.SkipIfNodeOSDistroIs("windows")` in this case. And can I add **B** in the second case in which `e2eskipper.SkipIfNodeOSDistroIs("linux")` is present.

Reply:
it seems it works correctly , disregard that comment please
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/107103/pull-kubernetes-e2e-aks-engine-windows-containerd/1472605943559622656/build-log.txt

let me recap,  the problem seems to be that we are conflating 3 networks tests in one test now, and 2 of them depend on an internet connection

1. ginkgo.It("should have stable networking for Linux and Windows pods", func() {
1.1 create a linux pod and check that I can reach www.google.com from the linux pod
1.2 create a windows pod and check that I can reach www.google.com from the windows pod
1.3 create a windows pod and a linux pod and check I can connect from one to the other and viceversa

what I suggest is that we split the connectivity tests, in a way that for airgap environments users can skip them

Reply:
you don't need additional skippers, just the `[Feature:Networking-DNS]`

```
		ginkgo.It("should provide Internet connection for containers using DNS [Feature:Networking-DNS]", func() {

			linuxPod := createTestPod(f, linuxBusyBoxImage, linuxOS)
			ginkgo.By("creating a linux pod and waiting for it to be running")
			linuxPod = f.PodClient().CreateSync(linuxPod)

			ginkgo.By("verifying pod external connectivity to the internet")

			ginkgo.By("checking connectivity to 8.8.8.8 53 (google.com) from Linux")
			assertConsistentConnectivity(f, linuxPod.ObjectMeta.Name, linuxOS, linuxCheck("8.8.8.8", 53))
```